File: pharmacy-ui/src/stateStore/Pharmacy/Pharmacy.ts
export interface Pharmacy {
    pharmacyId: number;
    name: string;
    city?: string;
    state?: string;
    zip?: string;
    filledPrescriptionMTD?: number;
    createdDate: Date;
    updatedDate?: Date;
}

export interface PharmacyState {
    loading: boolean;
    pharmacyList: Array<Pharmacy>;
    error: string | undefined;
  }

File: pharmacy-api/src/Nuvem.PharmacyManagement.PharmacyServices/DatabaseContext/EFEntities/Pharmacy.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Nuvem.PharmacyManagement.PharmacyServices.DatabaseContext.EFEntities;

public partial class Pharmacy
{
    [Key]
    public int PharmacyId { get; set; }

    [Required, MaxLength(200),MinLength(5)]
    public string Name { get; set; } = null!;

     [MaxLength(200),MinLength(5)]
    public string? Address { get; set; }

    [MaxLength(200),MinLength(3)]
    public string? City { get; set; }

     [MaxLength(200),MinLength(2)]
    public string? State { get; set; }

     [RegularExpression(@"^\d{5}(?:[-\s]\d{4})?$")]
    public string? Zip { get; set; }

    public int? RxFilledMtd { get; set; }

    public DateTime CreatedDate { get; set; }

    public DateTime? UpdatedDate { get; set; }
}
